src/raster/tests.rs

```diff
@@ -693,3 +693,38 @@ fn test_rasterize() {
         [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 0, 0, 0, 1, 1, 0, 0, 0, 0, 0, 0,]
     );
 }
+
+#[test]
+fn test_color_table() {
+    // Raster containing one band.
+    let dataset = Dataset::open(fixture!("test_color_table.tif")).expect("open failure");
+    assert_eq!(dataset.raster_count(), 1);
+
+    // Band is PaletteIndex.
+    let band = dataset.rasterband(1).expect("rasterband failure");
+    assert_eq!(
+        band.color_interpretation(),
+        ColorInterpretation::PaletteIndex
+    );
+
+    // Color table is RGB.
+    let color_table = band.color_table().unwrap();
+    assert_eq!(
+        color_table.palette_interpretation(),
+        gdal_sys::GDALPaletteInterp::GPI_RGB
+    );
+
+    // Color table has 256 entries.
+    let entry_count = color_table.entry_count();
+    assert_eq!(entry_count, 256);
+
+    // Check that entry and entry_as_rgb are the same.
+    for index in 0..entry_count {
+        let entry = color_table.entry(index).unwrap();
+        let rgb_entry = color_table.entry_as_rgb(index).unwrap();
+        assert_eq!(entry.c1, rgb_entry.c1);
+        assert_eq!(entry.c2, rgb_entry.c2);
+        assert_eq!(entry.c3, rgb_entry.c3);
+        assert_eq!(entry.c4, rgb_entry.c4);
+    }
```

Comment:
`GDALColorEntry` could work this way, although it is also from `gdal_sys`. But it would be maybe nicer to have a more readable struct. `c1` could be gray, _red, cyan or hue_  so it is not easy to name it. For the RGB part, we could split the structs to have one with r, g, b, a fields. What do you think?

Reply:
Maybe a `ColorEntry` enum that has the correct interpretation and we could forgo `palette_interpretation`?

Reply:
Have a look at the latest changes and let me know what you think.

Reply:
:+1: